src/AggregateFunctions/Streaming/AggregateFunctionDistinct.h

```diff
@@ -19,84 +19,130 @@ struct AggregateFunctionDistinctSingleNumericData
     using Self = AggregateFunctionDistinctSingleNumericData<T>;
     Set set;
 
-    /// proton: starts. Resolve multiple finalizations problem for streaming global aggreagtion query
+    /// Resolve multiple finalizations problem for streaming global aggreagtion query
+    /// Optimized, put the new coming data that the set does not have into extra_data_since_last_finalize.
     std::vector<T> extra_data_since_last_finalize;
-    bool use_extra_data = false;  /// Optimized, only streaming global aggreagtion query need to use extra data after first finalization.
-    /// proton: ends.
+
+    /// If has new data
+    bool has_new_data = false;
+
+    /// A blank mark that's used to decide which kind is the current block. 
+    /// 0 means data block, 1 means a special data block whose extra data has been deleted, 2 means a blank block.
+    int is_blank = 0;
 
     void add(const IColumn ** columns, size_t /* columns_num */, size_t row_num, Arena *)
     {
         const auto & vec = assert_cast<const ColumnVector<T> &>(*columns[0]).getData();
-        /// proton: starts.
         auto [_, inserted] = set.insert(vec[row_num]);
-        if (use_extra_data && inserted)
+        has_new_data = false;
```

Comment:
if only insert existed values ,  the flag `has_new_data` will be set false , is right ?

Reply:
Yes, the has_new_data only works when there is new data been inserted